### PR TITLE
HDDS-10699. Refactor ContainerBalancerTask and tests in TestContainerBalancerTask

### PR DESCRIPTION
**Pull request by Montura**
In [PR for HDDS-9889](https://github.com/apache/ozone/pull/5758) we [discussed](https://github.com/apache/ozone/pull/5758#pullrequestreview-1963251436) with [Siddhant Sangwan](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siddhant) that tests form `org.apache.hadoop.hdds.scm.container.balancer.TestContainerBalancerTask` could be refactored using `MockedSCM` class (introduced in [HDDS-9889](https://issues.apache.org/jira/browse/HDDS-9889))

## What changes were proposed in this pull request?

1. Refactor some tests from `org.apache.hadoop.hdds.scm.container.balancer.TestContainerBalancerTask` 
2. Made some cleanups in `org.apache.hadoop.hdds.scm.container.balancer.ContainerBalancerTask` related to changes in refactored test

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10699

## How was this patch tested?

Use standalone tests

### CODE REVIEW

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/MockedSCM.java [hunk 1]
```diff
@@ -137,7 +137,6 @@ public String toString() {
   }
 
   public @Nonnull ContainerBalancerTask startBalancerTask(@Nonnull ContainerBalancerConfiguration config) {
-    init(config);
```
Comment (Tejaskriya) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/MockedSCM.java:140]: In TestContainerBalancerDatanodeNodeLimit, each time init(config) is called, it is followed by startBalancerTask. We could retain this line and avoid calling mockedSCM.init() in every test case in TestContainerBalancerDatanodeNodeLimit.

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/TestContainerBalancerDatanodeNodeLimit.java [hunk 2]
```diff
@@ -79,6 +96,7 @@ public void containerBalancerShouldObeyMaxDatanodesToInvolveLimit(@Nonnull Mocke
     }
     config.setIterations(1);
     config.setMaxSizeToMovePerIteration(50 * STORAGE_UNIT);
+    mockedSCM.init(config);
```
Comment (Tejaskriya) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/TestContainerBalancerDatanodeNodeLimit.java:99]: Relating to the above comment, we would not require to call init here specifically. Similarly for all the test cases in this class.
What is the reason to remove init call within startBalancerTask?

Comment (Tejaskriya) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/TestContainerBalancerDatanodeNodeLimit.java]: Thanks, LGTM

**Review by sumitagrawl — APPROVED** [review]
@Montura Thanks for working over this, LGTM

**Comment by adoroszlai** [comment]
Thanks @Montura for the patch, @sumitagrawl, @Tejaskriya for the review.

**Pull request merged**